src/ConnectedMode/Migration/IConnectedModeMigration.cs

```diff
@@ -71,4 +62,49 @@ public MigrationProgress(int currentProject, int totalProjects, string message,
         /// </summary>
         public bool IsWarning { get; }
     }
+
```

Comment:
I've made the interface changes we discussed offline. We might need to tweak them later when we work on the implementation.